cross/toolchain.cmake

```diff
@@ -52,51 +43,25 @@ if(TARGET_ARCH_NAME STREQUAL "armel")
   endif()
 endif()
 
-# add_compile_param - adds only new options without duplicates.
-# arg0 - list with result options, arg1 - list with new options.
-# arg2 - optional argument, quick summary string for optional using CACHE FORCE mode.
-macro(add_compile_param)
-  if(NOT ${ARGC} MATCHES "^(2|3)$")
-    message(FATAL_ERROR "Wrong using add_compile_param! Two or three parameters must be given! See add_compile_param description.")
-  endif()
-  foreach(OPTION ${ARGV1})
-    if(NOT ${ARGV0} MATCHES "${OPTION}($| )")
-      set(${ARGV0} "${${ARGV0}} ${OPTION}")
-      if(${ARGC} EQUAL "3") # CACHE FORCE mode
-        set(${ARGV0} "${${ARGV0}}" CACHE STRING "${ARGV2}" FORCE)
-      endif()
-    endif()
-  endforeach()
-endmacro()
+set(CMAKE_SYSROOT "${CROSS_ROOTFS}")
+set(CMAKE_C_COMPILER_EXTERNAL_TOOLCHAIN "${CROSS_ROOTFS}/usr")
```

Comment:
If I understand it correctly, this replaces the following, right?
```cmake
  add_compile_options("--target=${TOOLCHAIN}")	
  add_compile_options("--gcc-toolchain=${CROSS_ROOTFS}/usr")
```
Before, it was Clang specific, now we do that for GCC too. I'm not sure it is correct.

Reply:
From my understanding of the CMAKE_<LANG>_EXTERNAL_TOOLCHAIN documentation, it's only used by compilers that don't ship a full toolchain with them ie Clang and not GCC)

Reply:
I've validated manually that these two flags are not passed to GCC.

Reply:
The sysroot is still passed, but the target and gcc-toolchain options are not passed.